### ISSUE TITLE
Failing the `checkPot` step should be more obvious.

### ISSUE THREAD

**Issue by feerrenrut** [nvaccess/nvda#10928]
### Is your feature request related to a problem? Please describe.
When the `checkPot` step fails it is not obvious leading to wasted time trying to find the error:
- When building the pot files all of the files checked are output to the command line, this can hide the following test failure.
- On Appveyor no message is presented on a PR and the build stops. The console output must be inspected to find the problem 

### Describe the solution you'd like
- Suppress the output of filenames when building pot files
- Run the `checkPot` step at the same time as other tests
- Inspect the `$LastExitcode` and use `Add-AppveyorMessage` to post a message to the PR
- Don't forget to fail the test with `if($errorCode -ne 0) { $host.SetShouldExit($errorCode) }`
  - See Unit tests section of `appveyor.yml` for example.

### Additional context
There may be some reason that the `checkPot` step has to be where it is that I'm not aware of.

**Comment by JCodyA** [comment]
Hello!  I'm looking for a good place to start contributing to NVDA and this looks promising.  I would love to help out with this one but I will definitely need guidance.  After a very brief initial look I can see where it looks like the filename is being printed https://github.com/nvaccess/nvda/blob/e8f14f9ae4d239fff28f78578c2a84128f214fa2/tests/checkPot.py#L224.  I'll keep looking through the code to try and learn more.

**Comment by feerrenrut** [comment]
Great! Thanks for your interest @JCodyA and welcome to the NVDA project. Please ask for help here if you would like guidance. There is also an [NVDA developers mailing list](https://groups.io/g/nvda-devel) if you aren't aware.

**Comment by CyrilleB79** [comment]
@seanbudd, does #12283 fix this issue? At least partially?
If this does not totally fix this issue, it should be clarified what remains to do.

**Comment by seanbudd** [comment]
> * Inspect the `$LastExitcode` and use `Add-AppveyorMessage` to post a message to the PR
> * Don't forget to fail the test with `if($errorCode -ne 0) { $host.SetShouldExit($errorCode) }`

These parts have been implemented on #12283

>Suppress the output of filenames when building pot files

I haven't fixed this - this would be a good to have so that the failures are included in the `tests` section. I think you have to structure the output a different way in the `checkPot` python script. 

>Run the checkPot step at the same time as other tests.

I think it was included earlier just to fail quicker - moving it would add ~5min to the AppVeyor process if it was to fail

**Comment by feerrenrut** [comment]
>>Run the checkPot step at the same time as other tests.
>I think it was included earlier just to fail quicker - moving it would add ~5min to the AppVeyor process if it was to fail

Yes, this is right. It's hard to know what is best. It would be good to know about the translation issues early, on the other hand a missing translation string prevents the author from knowing if the automated tests passed.

**Comment by kburtram** [comment]
Hi, I'm a developer that is interested in contributing to this project.  I saw this was a "good first issue" and took a shot at a proposed fix.  From reading through the issue I think there are two issues remaining (as far as I can tell).

1. Stop echo all the file names to the console build
2. Reorder the tasks in AppVeyor

I have a PR at https://github.com/kburtram/nvda/pull/2/files that has a proposed solution for this.  There is one commit for each of the two tasks above in case only one of the fixes is desired.  The basic proposed fix for 1 is to move the directory enumeration into one of the methods to avoid passing all the files to Env.Command, which is what is doing the echo'ing.

Please let me know if I should open a PR for this issue?  Or if there are any suggestions to changes I should do prior to opening a PR.  Thanks!

**Comment by seanbudd** [comment]
Hi @kburtram,

Thanks for investigating this and welcome to the NVDA repo! Feel free to just open PRs directly to the project, that way the community can provide feedback. I think in addition to these it would be good to ensure the failure information (i.e. lines that caused the failure) is handled by AppVeyor like a test failure so that you don't have to search the build log or download artefacts. Having these as individual PRs makes sense as well, as the reordering of tasks might be contentious.

**Comment by kburtram** [comment]
@seanbudd thanks for the suggestions and the welcome!  I'll split into two PRs to this repo for suppressing the console echoing and the AppVeyor changes.  And update the AppVeyor change to include the additional failure information.

**Comment by feerrenrut** [comment]
Hi @kburtram, thanks for taking a look.

>The basic proposed fix for 1 is to move the directory enumeration into one of the methods to avoid passing all the files to Env.Command, which is what is doing the echo'ing.

One thing to watch for here, is that SCons should still know to rebuild the pot file if the input files change.

**Comment by Adriani90** [comment]
@seanbudd, @michaelDCurran what needs to still be solved here? i see that at least the tests are looking for translation strings on PRs, but not sure if this is all fixed now.

**Comment by seanbudd** [comment]
@Adriani90 - It's in the same state as previous discussions in https://github.com/nvaccess/nvda/issues/10928#issuecomment-842740352 and https://github.com/nvaccess/nvda/issues/10928#issuecomment-842713989 which detail the remaining work

**Comment by seanbudd** [comment]
With GitHub actions, reviewing these test failures are a lot more simple. While XML Junit output would be helpful, there's not much value in generating it anymore
